src/ShapeFile.php

```diff
@@ -374,7 +374,7 @@ private function loadHeaders(): bool
         $this->readSHP(4);
 
         $shapeType = Util::loadData('V', $this->readSHP(4));
-        $this->shapeType = $shapeType === false ? -1 : (int) $shapeType;
+        $this->shapeType = ShapeType::tryFrom((int) $shapeType) ?? ShapeType::NULL;
```

Comment:
I think it's better to add an `Unknown` case instead of using `Null`, because `Null` is a valid type.
```suggestion
        $this->shapeType = ShapeType::tryFrom((int) $shapeType) ?? ShapeType::Unknown;
```